### PR TITLE
Add a project skeleton

### PR DESCRIPTION
**Pull request by connorkuehl**
This PR provides a basic makefile with a default target and a `clean` (`make clean`) target.

I've been building the project in a chroot like this:

```
$ schroot -c chroot:eoan-arm64 -- make
```

The Makefile doesn't modify the `CC` environment variable, so as long as your `CC` and `LD` environment variables are pointing at the right binaries it should work... I think.

I've been running it like this:

```
$ qemu-system-aarch64 -machine raspi3 -serial null -serial stdio -kernel kernel.elf
```

### CODE REVIEW

File: start.S [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+clear_bss:
+	/* Global variables live in the bss section, and therefore we must zero it
+	 * out.
+	 *
+	 * We've stored the size of the bss in w2, so once this hits zero, we
+	 * have cleared the entire section.
+	 */
+	cbz     w2, enter_kernel
+	str     xzr, [x1], #8
```
Comment (connorkuehl) [inline start.S:54]: Does this work or is this just writing a zero to the same place for this entire loop?

Comment (squidboylan) [reply start.S:54]: I think you're right and this just writes zero to the same place, i think if we swapped the `sub` and `str` and changed the `str` to be `str     xzr, [x1], w2` or something like that it'll do the right thing

File: linker.ld [hunk 2]
```diff
@@ -20,4 +20,4 @@ SECTIONS
 
    /DISCARD/ : { *(.comment) *(.gnu*) *(.note*) *(.eh_frame*) }
 }
-__bss_size = (__bss_end - __bss_start)>>3;
+__bss_size = (__bss_end - __bss_start)>>3; /* size of bss in BYTES */
```
Comment (squidboylan) [inline linker.ld]: wait if this is in bytes, that makes it sounds like bss_end and bss_start are in bits?

**Comment by squidboylan** [comment]
This LGTM, I'm happy to merge this, I have my docker stuff to add once this is in

File: start.S (hunk 1 above)
Comment (connorkuehl) [reply start.S:54]: https://github.com/bztsrc/raspi3-tutorial/issues/58#issuecomment-538686629

**Pull request merged**